drivers/serial/uart_gecko.c

```diff
@@ -0,0 +1,422 @@
+/*
+ * Copyright (c) 2017, Christian Taedcke
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <uart.h>
+#include <em_usart.h>
+#include <em_gpio.h>
+#include <em_cmu.h>
+#include <soc.h>
+
+struct uart_gecko_config {
+	USART_TypeDef *base;
+	CMU_Clock_TypeDef clock;
+	u32_t baud_rate;
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	void (*irq_config_func)(struct device *dev);
+#endif
+};
+
+struct uart_gecko_data {
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_callback_t callback;
+#endif
+};
+
+static int uart_gecko_poll_in(struct device *dev, unsigned char *c)
+{
+	const struct uart_gecko_config *config = dev->config->config_info;
+	u32_t flags = USART_StatusGet(config->base);
+	int ret = -1;
```

Comment:
you could avoid having a ret variable here:

Reply:
done